Pinta.Core/Classes/AsyncEffectRenderer.cs

```diff
@@ -117,11 +120,20 @@ internal double Progress {
 		}
 	}
 
+	private readonly object completion_swap_lock = new ();
 	internal void Start (
 		BaseEffect effect,
 		Cairo.ImageSurface source,
 		Cairo.ImageSurface dest)
 	{
+		TaskCompletionSource newCompletionSource = new ();
+		lock (completion_swap_lock) {
```

Comment:
Could you explain why the lock is needed here?

Not sure if I'm missing something, but if this is only called with `IsRendering == false` then it seems like there shouldn't be concurrent access to `completion_source` here?